File: eventhandler.py
class EventHandler:
	def __init__(self):
		self.bindings = {}
		
	def bind(self, eventname, function):
		eventbindings = self.bindings.setdefault(eventname, [])
		binding = {
			"event":    eventname,
			"function": function
		}
		eventbindings.append(binding)
		return binding
		
	def unbind(self, binding):
		self.bindings[binding['event']].remove(binding)
		
	def trigger(self, eventname, *args):
		eventbindings = self.bindings.setdefault(eventname, [])
		for binding in eventbindings:
			binding['function'](*args)

File: www/js/eventhandler.js
function EventHandler() {
		this.bindings = {};
		
		this.bind = function(eventname, fn) {
			var eventbindings = this.getBindings(eventname);
			var binding = {
				"event":	eventname,
				"fn":		fn
			}
			eventbindings.push(binding);
			return binding;
		}
		
		this.unbind = function(binding) {
			var eventname = binding.event;
			var eventbindings = this.bindings[eventname];
			var index = eventbindings.indexOf(binding);
			eventbindings.splice(index,1);
		}
		
		this.trigger = function(eventname, args) {
			var eventbindings = this.getBindings(eventname);
			for( var i=0; i<eventbindings.length; i++ ) {
				var binding = eventbindings[i];
				binding.fn.apply(null, args);
			}
		}
		
		this.getBindings = function(eventname) {
			if( !this.bindings[eventname] ) {
				this.bindings[eventname] = [];
			}
			var eventbindings = this.bindings[eventname];
			return eventbindings;
		}
}
